Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -640,93 +640,108 @@ def merge(args:)
         break if all_bottle
       end
 
-      if args.write?
-        if all_bottle
-          all_bottle_hash = nil
-          bottle_hash["bottle"]["tags"].each do |tag, tag_hash|
-            filename = Bottle::Filename.new(
-              formula_name,
-              bottle_hash["formula"]["pkg_version"],
-              tag,
-              bottle_hash["bottle"]["rebuild"],
-            )
-
-            if all_bottle_hash.nil?
-              all_bottle_tag_hash = tag_hash.dup
-
-              all_filename = Bottle::Filename.new(
-                formula_name,
-                bottle_hash["formula"]["pkg_version"],
-                "all",
-                bottle_hash["bottle"]["rebuild"],
-              )
-
-              all_bottle_tag_hash["filename"] = all_filename.url_encode
-              all_bottle_tag_hash["local_filename"] = all_filename.to_s
-              cellar = all_bottle_tag_hash.delete("cellar")
-
-              all_bottle_formula_hash = bottle_hash.dup
-              all_bottle_formula_hash["bottle"]["cellar"] = cellar
-              all_bottle_formula_hash["bottle"]["tags"] = { all: all_bottle_tag_hash }
-
-              all_bottle_hash = { formula_name => all_bottle_formula_hash }
-
-              puts "Copying #{filename} to #{all_filename}" if args.verbose?
-              FileUtils.cp filename.to_s, all_filename.to_s
-
-              puts "Writing #{all_filename.json}" if args.verbose?
-              all_local_json_path = Pathname(all_filename.json)
-              all_local_json_path.unlink if all_local_json_path.exist?
-              all_local_json_path.write(JSON.pretty_generate(all_bottle_hash))
-            end
+      unless args.write?
+        puts bottle_output(bottle)
+        next
+      end
 
-            puts "Removing #{filename} and #{filename.json}" if args.verbose?
-            FileUtils.rm_f [filename.to_s, filename.json]
-          end
-        end
+      path = Pathname.new((HOMEBREW_REPOSITORY/bottle_hash["formula"]["path"]).to_s)
+      formula = Formulary.factory(path)
+      old_bottle_spec = formula.bottle_specification
 
-        Homebrew.install_bundler_gems!
-        require "utils/ast"
+      no_bottle_changes = if old_bottle_spec &&
+                             bottle.rebuild  != old_bottle_spec.rebuild &&
+                             bottle.root_url == old_bottle_spec.root_url
```

Comment:
Do we perhaps want to compare pkg_version values? There is a couple formulae that do have version bumps that don't have any code changes (usually because it's a subproject of a larger one and they share version numbers), so it's theoretically possible for two versions to share checksums (albeit rare).